ext/sqlite3/sqlite3.c

```diff
@@ -117,7 +117,10 @@ PHP_METHOD(sqlite3, open)
 	if (strlen(filename) != filename_len) {
 		return;
 	}
-	if (filename_len != sizeof(":memory:")-1 ||
+	if (filename_len == 0) {
+		fullpath = emalloc(1);
```

Comment:
Why not, but you'll have to track that to be sure not to free it.
Otherwise : change the whole code to use a stack allocated buffer instead of a heap allocated one ;-)